lib/services/docker.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Package for cups service tests
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+package services::docker;
+use base 'opensusebasetest';
+use testapi;
+use utils;
+use strict;
+use warnings;
+use containers::docker;
+use containers::utils 'basic_container_tests';
+use containers::container_images 'build_and_run_image';
+
+my $service_type = 'Systemd';
+
+# check service is running and enabled
+sub check_service {
+    common_service_action 'docker', $service_type, 'is-enabled';
+    common_service_action 'docker', $service_type, 'is-active';
+}
+
+# check docker service before and after migration
+# stage is 'before' or 'after' system migration.
+sub full_docker_check {
+    my (%hash) = @_;
+    my ($stage, $type) = ($hash{stage}, $hash{service_type});
+    $service_type = $type;
+    my $engine = containers::docker->new();
+    $engine->init() if ($stage eq 'before');
+
+    check_service();
+
+    # Test the connectivity of Docker containers
+    $engine->check_containers_connectivity();
+
+    # Run basic runtime tests
+    basic_container_tests(runtime => $engine->runtime);
+    # Build an image from Dockerfile and run it
+    build_and_run_image(runtime => $engine, dockerfile => 'Dockerfile.python3', base => registry_url('python', '3'));
```

Comment:
based on the ticket description _There is NO need to test docker features_. i think it would be better to replace this with new things here. For instance, firewall checks, docker permission, filesystem checks or simple api calls like `docker images` etc

Reply:
@jlausuch @lemon-suse WDYT?

Reply:
That's a very good point.

I wrote the ticket thinking that migration tests don't actually test the feature itself, just the service, but I might be wrong.

I see some tests in https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/lib/services
that have a very basic checks in `check_function`, but some of them don't even implement this function...

Your code is definitely testing quite a lot compared to other migration tests, so IMO, I would leave only here `basic_container_tests` and remove `build_and_run_image`.

Reply:
Done.